python/lsst/daf/persistence/butler.py

```diff
@@ -61,6 +55,65 @@ class ButlerCfg(Policy, yaml.YAMLObject):
     def __init__(self, cls, repoCfg):
         super(ButlerCfg, self).__init__({'repoCfg':repoCfg, 'cls':cls})
 
+class RepoData(object):
+    def __init__(self, args, cfg, repo, tags):
+        self.args = args
+        self.cfg = cfg
+        self.repo = repo
+        self.mode = args.mode
+        # self.tags is used to keep track of *all* the applicable tags to the Repo, not just the tags in
+        # the cfg (e.g. parents inherit their childrens' tags)
+        if not isinstance(tags, set):
+            raise RuntimeError("tags passed into RepoData must be in a set.")
+        self.tags = tags
+
+    def __reduce__(self):
+        return (_unreduceRepoData, (self.args, self.cfg, self.repo, self.mode, self.tags))
+
+    def __repr__(self):
+        return "RepoData(args=%s cfg=%s repo=%s tags=%s" % (self.args, self.cfg, self.repo, self.tags)
+
+
+def _unreduceRepoData(args, cfg, repo, tags):
+    repoData = RepoData(RepositoryArgs(args, cfg, repo, tags))
+    return repoData
+
+
+class RepoDataContainer(object):
```

Comment:
Class and methods need docstrings.  I think conventions say there should be newlines between methods.